src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace System.Security.Permissions
+{
+    public sealed partial class PrincipalPermission : System.Security.IPermission, System.Security.ISecurityEncodable, System.Security.Permissions.IUnrestrictedPermission
+    {
+        public PrincipalPermission(System.Security.Permissions.PermissionState state) { }
+        public PrincipalPermission(string name, string role) { }
+        public PrincipalPermission(string name, string role, bool isAuthenticated) { }
+        public System.Security.IPermission Copy() { return default(System.Security.IPermission); }
+        public void Demand() { }
```

Comment:
> This one permission type is different from the rest: it doesn't derive from CodeAccessPermission and it's not really about Code Access Security.

I was wondering if the package was mis-described.  Guess I get to go leave that comment now :).

> but WindowsIdentity is currently only implemented for Windows, and while we will likely provide an implementation for at least some of it for Unix, we haven't yet.

Really? I hope not.  I know I'm apparently on the losing end of the debate here, but when things are clearly labelled as "Windows" they really shouldn't work on non-Windows (and vice-versa).

Though I guess that one is really a mis-named "ActiveDirectoryPrincipal", so maybe inasmuch as it makes sense to have a non-Windows machine joined to AD.

Anyways, this isn't the right place for this rant, but the email made me clench my jaw, and I had to vent.